implementations/rust/ockam/ockam_vault/src/service/models/asymmetric_request.rs

```diff
@@ -0,0 +1,54 @@
+use crate::service::models::PublicKeyModel;
+use minicbor::{Decode, Encode};
+use ockam_api::TypeTag;
+use ockam_core::compat::borrow::Cow;
+
+#[derive(Debug, Clone, Encode, Decode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct EcdhRequest<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<5767078>,
+    #[b(1)] secret_key_id: Cow<'a, str>,
+    #[b(2)] public_key: PublicKeyModel<'a>,
+}
+
+impl<'a> EcdhRequest<'a> {
+    pub fn new(secret_key_id: impl Into<Cow<'a, str>>, public_key: PublicKeyModel<'a>) -> Self {
+        Self {
+            tag: TypeTag,
+            secret_key_id: secret_key_id.into(),
+            public_key,
+        }
+    }
+    pub fn secret_key_id(&self) -> &Cow<'a, str> {
```

Comment:
I would hide the `Cow` as much as possible:
```suggestion
    pub fn secret_key_id(&self) -> &str {
```